### PR TITLE
Improvements to video functionality

### PR DESCRIPTION
**Pull request by Newnab**
This might need some wrangling because I was working from an RC1 version... 

There's a few things here:

1. Video thumbnails actually work now 😂 
2. I've added title to videos so you can specify a title/alt text rather than it always being blank
3. I've given the option of overriding the thumbnail with an image URL - ideally this would be a media picker in future but having something is better than nothing and the media picker stuff looked complicated. I'll make an issue for that once this is merged.
4. Rather than enforcing square thumbnails, I've made it so width/height can be set, and the defaults are 16:9 now which should match the new video default from YT. 
5. Output types on items by default so you can do e.g video specific styles (like a play icon)

### CODE REVIEW

File: assets/js/admin/components/galleryEditor/sources/videoEmbed/index.ts [hunk 1]
```diff
@@ -41,21 +55,22 @@ export default (id: string): IGallerySource => {
 
                         getEmbedThumb(url)
                             .then((thumb: string) => {
+                                console.log(thumb);
```
Comment (peterkeating) [inline assets/js/admin/components/galleryEditor/sources/videoEmbed/index.ts:58]: Let's remove this.

File: Models/GalleryPartItem.cs [hunk 2]
```diff
@@ -5,5 +5,6 @@ public class GalleryPartItem {
         public int Type { get; set; }
         public string TypeName { get; set; }
         public string Url { get; set; }
+        public string Thumb { get; set; }
```
Comment (peterkeating) [inline Models/GalleryPartItem.cs:8]: Ordering of properties within this class is alphabetical so I'd say move `Thumb` to top. When ordering class properties try to make them alphabetical so we have a consistent approach to ordering.

File: assets/js/admin/components/galleryEditor/sources/videoEmbed/index.ts (hunk 1 above)
Comment (Newnab) [reply assets/js/admin/components/galleryEditor/sources/videoEmbed/index.ts:58]: Good spot, totally accidental, sorry! 😬

**Comment by Newnab** [comment]
@peterkeating Was there anything else needed looking at here?

**Comment by peterkeating** [comment]
@Newnab Wanted to give it a run through, assume you're blocked by this so will try and do it today.

**Comment by peterkeating** [comment]
Created a new `rc1` branch as I know you'll need an rc1 build for the project you're working on. I'll also get these changes in to the latest RC2 version 👍

**Pull request merged**